5-appinfra/modules/cicd-pipeline/repository.tf

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  service_repo_name = var.cloudbuildv2_repository_config.repositories[var.repo_name].repository_name
+  use_csr           = var.cloudbuildv2_repository_config.repo_type == "CSR"
+}
+
+resource "google_sourcerepo_repository" "app_repo" {
+  count = local.use_csr ? 1 : 0
+
+
+  project = var.project_id
+  name    = var.repo_name
+
+  create_ignore_already_exists = true
+}
+
+resource "google_sourcerepo_repository_iam_member" "member" {
+  count = local.use_csr ? 1 : 0
+
+
+  project    = var.project_id
+  repository = google_sourcerepo_repository.app_repo[0].name
+  role       = "roles/source.admin"
+  member     = google_project_service_identity.cloudbuild_service_identity.member
+}
+
+module "cloudbuild_repositories" {
+  count = local.use_csr ? 0 : 1
```

Comment:
I'm thinking that it might be preferable to move the actual CSR creations to `test/setup`? That would keep this module generic, and then pass it through in a fashion similar to how customers would use it.